src/Microsoft.AspNet.Mvc.ModelBinding/Validation/DataAnnotationsModelValidator.cs

```diff
@@ -24,17 +24,20 @@ public bool IsRequired
 
         public IEnumerable<ModelValidationResult> Validate(ModelValidationContext validationContext)
         {
-            var metadata = validationContext.ModelMetadata;
+            var modelExplorer = validationContext.ModelExplorer;
+            var metadata = validationContext.ModelExplorer.Metadata;
+
             var memberName = metadata.PropertyName ?? metadata.ModelType.Name;
-            var containerMetadata = validationContext.ContainerMetadata;
-            var container = containerMetadata != null ? containerMetadata.Model : null;
-            var context = new ValidationContext(container ?? metadata.Model)
+            var containerExplorer = validationContext.ModelExplorer?.Container;
+
+            var container = containerExplorer?.Model;
+            var context = new ValidationContext(container ?? modelExplorer.Model)
             {
-                DisplayName = metadata.GetDisplayName(),
+                DisplayName = modelExplorer.Metadata.GetDisplayName(),
```

Comment:
nit: undo change to this line since you have a `metadata` variable with right value